File: app/src/main/java/com/cottondroid/todoachieved/auth/AuthenticationModule.java
package com.cottondroid.todoachieved.auth;

import com.google.firebase.auth.FirebaseAuth;

import dagger.Module;
import dagger.Provides;

@Module
public class AuthenticationModule {

    @Provides
    public FirebaseAuth provideFirebaseAuth() {
        return FirebaseAuth.getInstance();
    }

    @Provides
    public AuthenticationRepository authenticationRepository(FirebaseAuth firebaseAuth) {
        return new AuthenticationRepository(firebaseAuth);
    }
}

File: app/src/main/java/com/cottondroid/todoachieved/auth/AuthenticationModule.kt
package com.cottondroid.todoachieved.auth

import com.google.firebase.auth.FirebaseAuth
import dagger.Module
import dagger.Provides

@Module
class AuthenticationModule {
    @Provides
    fun provideFirebaseAuth(): FirebaseAuth {
        return FirebaseAuth.getInstance()
    }

    @Provides
    fun authenticationRepository(firebaseAuth: FirebaseAuth): AuthenticationRepository {
        return FirebaseAuthenticationRepository(firebaseAuth)
    }
}